operator/pkg/gateway-api/cell.go

```diff
@@ -86,8 +87,16 @@ func registerReconcilers(mgr ctrlRuntime.Manager, enableSecretSync bool, secrets
 	return nil
 }
 
-func registerGatewayAPITypesToScheme(scheme *runtime.Scheme) {
-	utilruntime.Must(gatewayv1.AddToScheme(scheme))
-	utilruntime.Must(gatewayv1beta1.AddToScheme(scheme))
-	utilruntime.Must(gatewayv1alpha2.AddToScheme(scheme))
+func registerGatewayAPITypesToScheme(scheme *runtime.Scheme) error {
+	for _, t := range []func(s *runtime.Scheme) error{
+		gatewayv1.AddToScheme,
+		gatewayv1beta1.AddToScheme,
+		gatewayv1alpha2.AddToScheme,
+	} {
+		if err := t(scheme); err != nil {
+			return fmt.Errorf("failed to add types to scheme: %w", err)
```

Comment:
(nit) maybe `failed to add type %t to scheme: %w` if there's some way to get a human readable name of the type?

(also applies to the registration in `cell.go`)

Reply:
Would be nice, but since its only a function the output of `%T` would be that much of help `func(*runtime.Scheme) error`

Reply:
What about:

```go
for gv, f := range map[fmt.Stringer]func(s *runtime.Scheme) error{
	gatewayv1.GroupVersion:       gatewayv1.AddToScheme,
	gatewayv1beta1.GroupVersion:  gatewayv1beta1.AddToScheme,
	gatewayv1alpha2.GroupVersion: gatewayv1alpha2.AddToScheme,
} {
	if err := f(scheme); err != nil {
		return fmt.Errorf("failed to add types from %s to scheme: %w", gv, err)
	}
}
```

Anyway, not a blocker, I'll leave this to you.

Reply:
Thanks for the idea @pippolo84 . This works if the package provides the group version as global variable. This is not the case for `clientgoscheme` (aggregation of multiple groupversions) that gets added to the scheme in the `controller-runtime` cell. IMO this is why we should rely on the underlying error to provide enough information for analysing any potential issue.

Reply:
Changed